src/error.rs

```diff
@@ -132,4 +132,11 @@ impl fmt::Display for Error {
     }
 }
 
-impl error::Error for Error {}
+impl error::Error for Error {
+    fn source(&self) -> Option<&(dyn error::Error + 'static)> {
+        match self.kind {
+            Kind::Io(ref e) => Some(e),
```

Comment:
I always find this part a little tricky. I *think* since the `h2::Error` pretty much *is* the `io::Error`, this should instead be `e.source()`, similar to how `io::Error` works with a custom error.

Reply:
Hm. The specific reason why I want this is so that I can traverse a `&dyn Error`'s cause chain and discover an `io::Error`; right now I have to also check for an `h2::Error` explicitly, even if that error is an IO error.